client/lib/tree-convert/test/index.js

```diff
@@ -0,0 +1,74 @@
+/**
+ * External dependencies
+ */
+import { expect } from 'chai';
+import cloneDeep from 'lodash/cloneDeep';
+
+/**
+ * Internal dependencies
+ */
+import { menusFlat, menuBadParent } from './fixtures';
+import TreeConvert from '../index';
+
+describe( 'TreeConvert', () => {
+	let converter, menu;
+
+	beforeEach( () => {
+		converter = TreeConvert();
+		menu = cloneDeep( menusFlat.menus[0] );
+	} );
+
+	describe( 'treeify', () => {
+		it( 'should turn a flat parent/child structure into a tree', () => {
+			const treeified = converter.treeify( menu.items );
+
+			expect( treeified ).to.have.length( 3 );
+			expect( treeified[2].items ).to.be.instanceOf( Array );
+		} );
+
+		it( 'should order sibling items by their "order" property', () => {
+			const treeified = converter.treeify( menu.items );
+
+			expect( treeified[1].items[0] ).to.have.property( 'name', 'Socks' );
+		} );
+
+		it( 'should remove the "order" property from items, since order will be implicit', () => {
+			const treeified = converter.treeify( menu.items );
+
+			expect( treeified[1].items[0] ).not.to.have.property( 'order' );
+		} );
+
+		it( 'should handle bad parent data, falling back to parent 0', () => {
+			const badMenu = cloneDeep( menuBadParent );
```

Comment:
I don't think your prior explanation was off. _I think_ this is another habit of mine. We've been enforcing one declaration per line.

Just to make sure I got this, the correct `const` usage here would be: 
```
const badMenu = cloneDeep( menuBadParent ), treeified = converter.treeify( badMenu.items );
// or
const badMenu = cloneDeep( menuBadParent ) ,
  treeified = cloneDeep( badMenu.items );
```

And the same applies to `let` I'm assuming. Extending your example:

```
const a = 1, b = 2, c = 3;
let i = 0, j = 0; 
...